futures-core/src/macros/pin.rs

```diff
@@ -0,0 +1,43 @@
+#[macro_export]
+macro_rules! unsafe_pinned {
+    ($f:tt -> $t:ty) => (
+        fn $f<'a>(
+            self: &'a mut $crate::core_reexport::mem::PinMut<Self>
+        ) -> $crate::core_reexport::mem::PinMut<'a, $t> {
+            unsafe {
+                $crate::core_reexport::mem::PinMut::map_unchecked(
+                    self.reborrow(), |x| &mut x.$f
+                )
```

Comment:
I've changed this to `map_unchecked`. It works the same.

Also, here and in the following macros the `$crate::core_reexport::` prefixes were partly missing.